### PR TITLE
System Transformation - Automatically make PDESystems compatible

### PR DESCRIPTION
**Pull request by xtalax**
Allows
- Mixed derivatives 
- Composite derivative arguments
- Time agnostic BC/IC parsing
- More free choice of ic equation structure
- Arbitrary order time derivatives
- Expandable derivative terms in equations
- Use ODAEProblem where appropriate

`pde_system_tranformation.jl` is where the meat of this happens

### CODE REVIEW

**Comment by xtalax** [comment]
@ChrisRackauckas Ok, this is actually "working" in that it's correctly transforming the system and addig the right new equations, but it's making solve time oppressively long.

I'm testing on periodic `Dt(u) ~ Dxy(u)`, but this is unstable even with WENO and sensibly short timesteps. Can you suggest a better system to test?

File: src/MOL_discretization.jl [hunk 1]
```diff
@@ -61,8 +63,8 @@ function SciMLBase.symbolic_discretize(pdesys::PDESystem, discretization::Method
     ####
     for pde in pdeeqs
         # Read the dependent variables on both sides of the equation
-        depvars_lhs = get_depvars(pde.lhs, depvar_ops)
-        depvars_rhs = get_depvars(pde.rhs, depvar_ops)
+        depvars_lhs = get_depvars(pde.lhs, v.depvar_ops)
+        depvars_rhs = get_depvars(pde.rhs, v.depvar_ops)
```
Comment (ChrisRackauckas) [inline src/MOL_discretization.jl:67]: why not flatten here?

Comment (xtalax) [reply src/MOL_discretization.jl:67]: Can you expand? This code is inherited from diffeqops

**Comment by ChrisRackauckas** [comment]
lgtm

File: test/pde_systems/system_transformation.jl [hunk 2]
```diff
@@ -0,0 +1,2 @@
+using MethodOfLines, ModelingToolkit, OrdinaryDiffEq, DomainSets
+using Test
```
Comment (ChrisRackauckas) [inline test/pde_systems/system_transformation.jl:2]: This is dead code?

Comment (xtalax) [reply test/pde_systems/system_transformation.jl:2]: ah good catch, I had been running it direct I'll add to runtests

**Comment by xtalax** [comment]
CI failures work locally, merging to get on with other stuff

**Pull request merged**